extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -413,6 +416,12 @@ export class AmpA4A extends AMP.BaseElement {
                 this.win.ampA4aValidationKeys = this.getKeyInfoSets_();
                 return this.verifyCreativeSignature_(
                     creativeParts.creative, creativeParts.signature);
+              })
+              // Emit adResponseValidateEnd for any non-erroneous validation
+              // completion, but only if we've actually started validation.
+              .then(creative => {
```

Comment:
There probably is some cost, you're right.  I was trying to keep most pings inside the chain, but that's probably hopeless.  Moved.